app/hooks/useLanguage.tsx

```diff
@@ -0,0 +1,38 @@
+import type { Dispatch, FunctionComponent, PropsWithChildren, SetStateAction } from 'react'
+import { createContext, useContext, useEffect, useState } from 'react'
+
+export enum Language {
+  EN = 'en-US',
+  DE = 'de-DE'
+}
+
+export const DEFAULT_LANGUAGE = Language.DE
+
+export interface LanguageContextValue {
+  language: Language,
+  setLanguage: Dispatch<SetStateAction<Language>>
+}
```

Comment:
Should probably decide on using either `type` or `interface` for most things, I personally much prefer `type` as it gives a lot more freedom with things like union and intersection types.